runtime/etiao/src/main/java/quarks/runtime/etiao/EtiaoJob.java

```diff
@@ -51,6 +53,10 @@ public EtiaoJob(DirectGraph graph, String topologyName, ServiceContainer contain
         ControlService cs = container.getService(ControlService.class);
         if (cs != null)
             cs.registerControl(JobMXBean.TYPE, getId(), getName(), JobMXBean.class, new EtiaoJobBean(this));
+        
+        this.jobs = (JobRegistry) container.getService(JobRegistryService.class);
```

Comment:
Maybe then it's the sole responsibility of the runtime to register a JobRegistry and thus JobRegistry would not be a public class, but instead an implementation detail

Then it would not be expected that external code registers a JobRegistryService .